### PR TITLE
[v7r0] Allow extensions to extend CI tests

### PR DESCRIPTION
**Pull request by chrisburr**
Modifies the CI to support LHCbDIRAC's CI tests.
Targets `rel-v7r0` so it can go in to LHCbDIRAC v10r0.

BEGINRELEASENOTES

*Tests
NEW: Allow extensions to extend the continuous integration tests

ENDRELEASENOTES

### CODE REVIEW

File: tests/Integration/all_integration_server_tests.sh [hunk 1]
```diff
@@ -85,4 +81,5 @@ echo -e "*** $(date -u)  **** Resources TESTS ****\n"
 python $SERVERINSTALLDIR/DIRAC/tests/Integration/Resources/Storage/Test_Resources_GFAL2StorageBase.py ProductionSandboxSE 2>&1 | tee -a $SERVER_TEST_OUTPUT; (( ERR |= $? ))
 
 # Can only run if there's a Stomp MQ local...
+# TODO Enable
 # python -m pytest $SERVERINSTALLDIR/DIRAC/tests/Integration/Resources/MessageQueue/Test_ActiveClose.py 2>&1 | tee -a $SERVER_TEST_OUTPUT; (( ERR |= $? ))
```
Comment (chrisburr) [inline tests/Integration/all_integration_server_tests.sh]: Should I open an issue to add another service to the docker-compose file so this test can be ran? Or is there a reason to not run it?

File: tests/Jenkins/dirac_ci.sh [hunk 2]
```diff
@@ -130,18 +133,23 @@ function installSite(){
...
+      else
+        option+="${module_path},"
+      fi
+    done
+    INSTALLOPTIONS+=("${option: :$((${#option} - 1))}")
   fi
 
-  echo "==> Installing with options $installOptions $SERVERINSTALLDIR/install.cfg"
+  # HACK: dirac-install.py currently puts installation symlinks in the current working directory
+  cd "${SERVERINSTALLDIR}" || exit 1
```
Comment (chrisburr) [inline tests/Jenkins/dirac_ci.sh]: This is a workaround for https://github.com/DIRACGrid/DIRAC/pull/4430

File: tests/CI/install_client.sh [hunk 3]
```diff
@@ -13,42 +13,43 @@ set -x
...
-
-DIRACSETUP=$(< tests/Jenkins/install.cfg grep "Setup = " | cut -f5 -d " ")
-
-cd ../..
-
+for repo_path in "${TESTREPO[@]}"; do
+    if [[ -d "${repo_path}" ]]; then
+        cp -r "${repo_path}" "$(basename "${repo_path}")"
+        cd "$(basename "${repo_path}")"
+        echo "Using local test repository in branch $(git branch | grep "\*" | sed -e "s/*^* //")"
```
[the remark this answers is not in the document]
Comment (chrisburr) [reply tests/CI/install_client.sh]: I'm not sure exactly what this does in sed's default pattern matching engine but changing it results in something different happening.
```
$ echo -e "*\n**** Hello" | sed -e "s/*^* //"
*
***Hello
$ echo -e "*\n**** Hello" | sed -e "s/^* //"
*
**** Hello
$ echo -e "*\n**** Hello" | sed -e "s/*^ //"
*
**** Hello
$ echo -e "*\n**** Hello" | sed -e "s/** //"
*
Hello
```

As this is inherited from what was already here I'm inclined to leave it as is.

[the remark this answers is not in the document]
Comment (andresailer) [reply tests/CI/install_client.sh]: Hmm, the problems of regular expressions :) (https://xkcd.com/1171/)

Thinking harder:

`*^* ` means a star followed by zero or more circumflexes followed by a space. (Star at the beginning of an expression is literal *, circumflex _not_ at the beginning is the character itself, followed by the star operator. Circumflexes cannot show up in branch names however (I think).

Then, assuming `git branch` doesn't change its output, one would only ever have `* branchName`, never more than one star. Then one could also just replace `* ` (star followed by space)...

File: tests/Integration/all_integration_server_tests.sh (hunk 1 above)
Comment (fstagni) [reply tests/Integration/all_integration_server_tests.sh]: No, as right now I am not even sure this test can run at all.

File: .github/workflows/basic.yml [hunk 4]
```diff
@@ -18,6 +18,9 @@ jobs:
           - DIRAC_USE_M2CRYPTO=Yes pytest Core/Security/test
           - tests/checkDocs.sh
           # TODO This should cover more than just tests/CI
+          # Excluded codes relared to sourcing files
```
Comment (andresailer) [inline .github/workflows/basic.yml]:
```suggestion
          # Excluded codes related to sourcing files
```

File: tests/Jenkins/dirac_ci.sh (hunk 2 above)
Comment (fstagni) [reply tests/Jenkins/dirac_ci.sh]: So, if #4430 is merged, will you rebase and remove this hack?
Now, in https://github.com/DIRACGrid/DIRAC/pull/4436#issuecomment-583028471 there's a comment about another bug fix that should be in this same PR, and for which @andresailer committed yet another "hack/temp fix" (https://github.com/DIRACGrid/DIRAC/pull/4436/commits/1c342128472cad2cd1d17e54ddc99bd9eceb1733) that will be removed after.
So, can you please add to #4430 the fix mentioned above? Then I will merge, propagate, and both of you can fix your respective PRs.

Comment (chrisburr) [reply tests/Jenkins/dirac_ci.sh]: Okay. I've applied the commit to #4430 and the CI has passed.

**Comment by fstagni** [comment]
Rebase now

**Comment by fstagni** [comment]
Are you still working on this, or is it ready for being merged?

**Comment by chrisburr** [comment]
I think this is done so go ahead and merge. If I want to add anything it’s better for me to make an easier to review PR.

**Comment by atsareg** [comment]
@andresailer: Approved ?

**Pull request merged**